### DISCUSSION TITLE
Hierarchies Considered Harmful

### DISCUSSION THREAD

**Discussion by wagle** [SkepticMystic/breadcrumbs#395] [Q&A]
I've spent half my life failing to shoehorn my mind into hierarchies.  It.  Does.  Not.  Work.

I often have "ten" different reasons to do anything.  There is no preferred "parent".  Making the wrong adhoc shoehorning decision makes the child inaccessible to the other "parents", and it is lost.

In real life, I have two parents, each having two more parents, ad infinitum.  I have one sibling, and we might each have 1 spouse, having their own ancestry, 2 children, again ad infinitum.

I was attracted to obsidian after a very long search because it represented my DAG's, which are huge and so hard to reason about globally.  Many people appear to have no problem shoehorning their minds into hierarchy/outlines, and so have no problem insisting that I do the same.  It can't be done.

I'm looking into juggl, which seems to do DAG's, but it uses breadcrumbs, so I'm trying to do multiple parents in it.

I don't see the glory in reducing a DAG to a TREE.  Please explain?

**Comment by wagle** [comment]
For example, in your README.md at [https://github.com/SkepticMystic/breadcrumbs](https://github.com/SkepticMystic/breadcrumbs), you say:

```
Frontmatter field. This is a field in the yaml format at the very top of your note. For example the me.md file could have the following frontmatter:

---
parent: [[dad]]
parent: [[mom]]
sibling: [[sister Lara]]
child: [[child Andre]]
---
```

Note the multiple parents?

**Comment by wagle** [comment]
I do seem to be trying to impose a dag (instead of a tree) on the cyclic obsidian graph.

**Comment by neyer** [comment]
Maybe I'm confused here, how are DAGS not hierarchies?

I agree that trees aren't great, since they're too inflexible. But i'd consider a dag to be hierarchical in that if you 'keep going up' you'll end up finding _some_ root or set of roots.

**Reply by wagle** [reply]
Hierarchies are trees.  Each non-root node has a unique parent.  If a node is a task, then there can be only one reason to do it.  In a DAG, you can have the same task serve multiple purposes.

**Comment by michaelpporter** [comment]
Breadcrumbs isn't limited to trees — it builds a directed graph, so a DAG is exactly what it represents. A note can have multiple parents (`up: ["[[mom]]", "[[dad]]"]`), and because edge fields are arbitrary and user-defined, you aren't forced onto a single "parent" axis at all: model as many independent relation types as you like, and a child stays reachable from every one of them.

(YAML aside: you can't repeat the same key twice — `parent:` then `parent:` — use a list under one key as above. The duplicate-key example in the old SkepticMystic README is v3-era docs; v4 is frontmatter-list based.)

Leaving this open as a discussion — thanks for the thoughtful write-up.
